tests/fixtures/search.py

```diff
@@ -38,41 +38,31 @@ class ExternalSearchFixture(LoggerMixin):
     to ensure that it works well overall, with a realistic index.
     """
 
-    def __init__(self, db: DatabaseTransactionFixture, services: Services):
+    def __init__(
+        self, db: DatabaseTransactionFixture, services: Services, test_id: TestIdFixture
+    ):
         self.search_config = SearchTestConfiguration()
         self.services_container = services
+        self.index_prefix = test_id.id
 
         # Set up our testing search instance in the services container
         self.search_container = Search()
-        self.search_container.config.from_dict(self.search_config.dict())
+        search_config_dict = self.search_config.dict()
+        search_config_dict["index_prefix"] = self.index_prefix
+        self.search_container.config.from_dict(search_config_dict)
         self.services_container.search.override(self.search_container)
 
-        self._indexes_created: list[str] = []
```

Comment:
Previously this fixture tracked all the indexes created and then deleted them. Now it gets a unique index prefix per fixture invocation, and all indexes are created with this prefix. When the fixture is cleaned up, all the indexes with that prefix are deleted. 

This allows the fixture to run in parallel without collisions between names, and makes the cleanup logic easier and less likely to leak an index.